### PR TITLE
Call Prometheus, Loki and Jaeger via Grafana API

### PR DESCRIPTION
**Pull request by dennis-ge**
**Description**

Changes proposed in this pull request:

- Calling prometheus, jaeger, and loki via the Grafana datasource API (see `proxyGrafanaDashboard`). 
    - By doing this, we get rid of kubectl forward (see `kubectlportForward`), and execute the integration tests more from a user perspective. 
    - The initial idea was to call the services via the apiserver proxy. This worked for prometheus but didn't work for loki and jaeger as istio `authorizationpolicies` and `peerauthorization` resources restricted the access to these services.
- As a consequence, Grafana needs to be exposed before the first call to any datasource (prometheus, jaeger, and loki) is triggered. That's why the monitoring tests are now executed first in the standard test suite. The Grafana proxy is then reset at the end of the test.
- Fixing type coercion
- Use the new logging functions (`info`, `error`) in the observability tests
- Extract Jaeger client calls to `tracing` directory

Pipelines:
- https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/dennis-ge_test_of_prowjob_kyma-gke-nightly-fast-integration
- https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/dennis-ge_test_of_prowjob_kyma-integration-k3d
- https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/dennis-ge_test_of_prowjob_kyma-upgrade-k3d-kyma2-to-main
- https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/dennis-ge_test_of_prowjob_post-main-kyma-gardener-azure-alpha-e

**Related issue(s)**
Fixes https://github.com/kyma-project/kyma/issues/13375

### CODE REVIEW

**Review by VladislavPaskar — APPROVED** [review]

File: tests/fast-integration/monitoring/grafana.js [hunk 1]
```diff
@@ -15,75 +14,74 @@ const {
...
-  if (redirectURL.includes('https://dex.')) {
-    console.log('Checking redirect for dex');
-    return await retryUrl(url, redirectURL, ignoreSSL, 200);
-  }
-
-  if (redirectURL.includes('https://kyma-project.io/docs')) {
-    console.log('Checking redirect for kyma docs');
-    return await retryUrl(url, redirectURL, ignoreSSL, 403);
+async function setGrafanaProxy() {
+  if (getEnvOrDefault('KYMA_MAJOR_VERSION', '2') === '2') {
```
Comment (skhalash) [inline tests/fast-integration/monitoring/grafana.js:67]: Do you think we still need it? I thought all Kyma 1.* remnants are gone

File: tests/fast-integration/monitoring/grafana.js [hunk 2]
```diff
@@ -15,75 +14,74 @@ const {
...
-  const url = 'https://' + hosts;
-  if (redirectURL.includes('https://dex.')) {
-    console.log('Checking redirect for dex');
-    return await retryUrl(url, redirectURL, ignoreSSL, 200);
-  }
-
-  if (redirectURL.includes('https://kyma-project.io/docs')) {
-    console.log('Checking redirect for kyma docs');
-    return await retryUrl(url, redirectURL, ignoreSSL, 403);
+async function setGrafanaProxy() {
```
Comment (skhalash) [inline tests/fast-integration/monitoring/grafana.js:66]: IMHO, the function name does not describe its purpose. Maybe something like `exposeGrafana`\`unexposeGrafana`?

File: tests/fast-integration/monitoring/grafana.js [hunk 3]
```diff
@@ -15,75 +14,74 @@ const {
...
+  if (getEnvOrDefault('KYMA_MAJOR_VERSION', '2') === '2') {
+    await manageSecret('create');
+    await restartProxyPod();
+    await updateProxyDeployment('--reverse-proxy=true', '--trusted-ip=0.0.0.0/0');
+    info('Checking grafana redirect to grafana URL');
+    const res = await checkGrafanaRedirect('https://grafana.', 200);
+    assert.isTrue(res, 'Grafana redirect to grafana landing page does not work!');
```
Comment (skhalash) [inline tests/fast-integration/monitoring/grafana.js:73]: I don't think we should assert inside helper functions? Maybe throw an exception?

File: tests/fast-integration/monitoring/grafana.js [hunk 4]
```diff
@@ -93,35 +91,32 @@ async function manageSecret(action) {
     kind: 'Secret',
```
Comment (skhalash) [inline tests/fast-integration/monitoring/grafana.js:91]: Could you please refactor this function (split into `createProxySecret`, `deleteProxySecret`)?

File: tests/fast-integration/eventing-test/eventing-test.js [hunk 5]
```diff
@@ -70,6 +74,10 @@ describe('Eventing tests', function() {
     await waitForNamespace(mockNamespace);
   });
 
+  if (!isSKR) {
+    setGrafanaProxy();
+  }
```
Comment (skhalash) [inline tests/fast-integration/eventing-test/eventing-test.js:79]: Why is it only enabled on OS? Please check if you could set/reset into the mocha hooks (before/after)?

File: tests/fast-integration/package-lock.json [hunk 6]
```diff
@@ -242,9 +242,9 @@
       "dev": true
     },
     "ansi-regex": {
-      "version": "3.0.1",
-      "resolved": "https://registry.npmjs.org/ansi-regex/-/ansi-regex-3.0.1.tgz",
-      "integrity": "sha512-+O9Jct8wf++lXxxFc4hc8LsjaSq0HFzzL7cVsw8pRDIPdjKD2mT4ytDZlLuSBZ4cLKZFXIrMGO7DbQCtMJJMKw=="
+      "version": "3.0.0",
+      "resolved": "https://registry.npmjs.org/ansi-regex/-/ansi-regex-3.0.0.tgz",
+      "integrity": "sha1-7QMXwyIGT3lGbAKWa922Bas32Zg="
```
Comment (skhalash) [inline tests/fast-integration/package-lock.json:247]: Seems to be an accidental change

**Review by skhalash — CHANGES_REQUESTED** [review]

File: tests/fast-integration/monitoring/grafana.js (hunk 1 above)
Comment (dennis-ge) [reply tests/fast-integration/monitoring/grafana.js:67]: I would keep it for now as it is still used in other places in the code and remove them all in one follow-up PR.

File: tests/fast-integration/monitoring/grafana.js (hunk 3 above)
Comment (dennis-ge) [reply tests/fast-integration/monitoring/grafana.js:73]: We also assert in other helper functions

Comment (skhalash) [reply tests/fast-integration/monitoring/grafana.js:84]: Still think it hinders test glanceability, but let's keep it for now

**Pull request merged**